principles.md

```diff
@@ -0,0 +1,25 @@
+## Product and Engineering Principles
+
+**Simpler always wins in the end**
+
+We choose simpler designs whenever possible. Having less configuration knobs
+is better for security. Complex products are easier to hack and harder
+to use. Security feature is only good, when it's easy to use.
+
+That's why we stick to [client certificates](https://goteleport.com/teleport/how-it-works/certificate-based-authentication-ssh-kubernetes/)
+and as the only way to authenticate and [reduce the number of ports and protocols](https://github.com/gravitational/teleport/issues/5777).
+
+**Security through transparency**
+
+The only way to build secure systems is through peer review and transparency.
+
+That's why we have [open design process](https://github.com/gravitational/teleport/pull/5135/files),
+and [publish security audits](https://blog.doyensec.com/2020/03/02/gravitational-audit.html) even
+if the reports are not always shiny.
+
+**Security and customers are not the product**
+
+Everyone deserves strong security and privacy.
+
+That is why all security features in Teleport are free and our software does not spy on our customers.
```

Comment:
One might argue that SSO is a security feature https://sso.tax/

Reply:
Sure, that's one way of looking at it and the boundary is blurry. I always viewed SSO and RBAC an organization scalability tool. From the security perspective, strong encryption, strong password and second factor are as secure as you can get.  Lots of vulnerabilities we have observed were actually caused by SAML and provider integration.

Reply:
```suggestion
That's why the free version of Teleport is built to be fully secure and our software does not spy on our customers.
```

Reply:
Encryption and password+2fa auth don't save you from lingering accounts of former employees still being usable long after those employees left.
It is a social vulnerability rather than a technical one, so you could wiggle out of that argument.

Reply:
All makes sense, we don't have to declare it as a principle if we don't think it's strictly true or does not reflect the reality. Let me think of some alternatives, and if you have ideas, please let me know